README.rst

```diff
@@ -153,19 +147,92 @@ See the full `Chat Message History`_ tutorial.
 
 .. _`Chat Message History`: https://github.com/googleapis/langchain-google-alloydb-pg-python/tree/main/docs/chat_message_history.ipynb
 
+Example Usage
+-------------
+
+Code examples can be found in the `samples/`_ folder.
+
+.. _samples/: https://github.com/googleapis/langchain-google-alloydb-pg-python/tree/main/samples
+
+Converting between Sync & Async Usage
+-------------------------------------
+
+Async functionality improves the speed and efficiency of database connections through concurrency,
+which is key for providing enterprise quality performance and scaling in GenAI applications. This
+package uses a native async Postgres driver, `asyncpg`_, to optimize Python's async functionality.
+
+LangChain supports `async programming`_, since LLM based application utilize many I/O-bound operations,
+such as making API calls to language models, databases, or other services. All components should provide
+both async and sync versions of all methods.
+
+`asyncio`_ is a Python library used for concurrent programming and is used as the foundation for multiple
+Python asynchronous frameworks. asyncio uses `async` / `await` syntax to achieve concurrency for
+non-blocking I/O-bound tasks using one thread with cooperative multitasking instead of multi-threading.
+
+.. _`async programming`: https://python.langchain.com/docs/concepts/async/
+.. _`asyncio`: https://docs.python.org/3/library/asyncio.html
+.. _`asyncpg`: https://github.com/MagicStack/asyncpg
+
+Converting Sync to Async
```

Comment:
Should we include Async to Sync as well? Or is that overkill?

Reply:
That feels a bit overkill